sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/FieldsShouldNotBePublic.cs

```diff
@@ -63,12 +63,12 @@ protected sealed override void Initialize(SonarAnalysisContext context)
 
                     var firstVariable = fieldDeclaration.Declaration.Variables[0];
                     var symbol = c.SemanticModel.GetDeclaredSymbol(firstVariable);
-                    if (symbol == null || !symbol.IsPublicApi())
+
+                    if (symbol != null &&
+                        symbol.GetEffectiveAccessibility() == Accessibility.Public)

```

Comment:
You could remove the previous check, GetEffectiveVisibility will handle the null